### DISCUSSION TITLE
[v4] Vite can't find "./tailwind.css"

### DISCUSSION THREAD

**Discussion by Stanzilla** [tailwindlabs/tailwindcss#14878] [Help]
**What version of Tailwind CSS are you using?**

4.0.0-alpha.30

**What build tool (or framework if it abstracts the build tool) are you using?**

Vite 5.4.10

**What version of Node.js are you using?**

22.9.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**Reproduction URL**

Not possible to generate with Vite

**Describe your issue**

Getting the following error

```
[plugin:vite:import-analysis] Missing "./tailwind.css" specifier in "tailwindcss" package
9:55:00 PM [vite] Pre-transform error: Package path ./tailwind.css is not exported from package /home/stan/projects/personal/wago-app/node_modules/tailwindcss (see exports field in /home/stan/projects/personal/wago-app/node_modules/tailwindcss/package.json)
```

my app.css

```css
@import 'tailwindcss';
@config '../../tailwind.config.mjs';
```

my tailwind config

```js
/** @type { import('tailwindcss').Config } */

const config = {
  content: ['./index.html', './src/**/*.{js,jsx,ts,tsx}'],
  darkMode: 'media',
  theme: {
    extend: {
      backgroundImage: {
        'gradient-overlay':
          'linear-gradient(to left, rgba(0,0,0,0.5), rgba(0,0,0,0))',
      },
      fontFamily: {
        sans: [
          'Inter',
          'ui-sans-serif',
          'system-ui',
          '-apple-system',
          'BlinkMacSystemFont',
        ],
      },
      typography: {
        DEFAULT: {
          css: {
            color: '#FFF',
            h1: {
              color: '#FFF',
            },
            h2: {
              color: '#FFF',
            },
          },
        },
      },
      colors: {
        'wago-red': '#C1272D',
        'wago-red-hover': '#D6262D',
        'wago-yellow': {
          DEFAULT: '#F5BB00',
          50: '#FFF7DB',
          100: '#FFF1C2',
          200: '#FFE48F',
          300: '#FFD85C',
          400: '#FFCC29',
          500: '#F5BB00',
          600: '#C29400',
          700: '#8F6D00',
          800: '#5C4600',
          900: '#291F00',
        },
        'wago-light-blue': {
          DEFAULT: '#73D2DE',
          50: '#D0F0F4',
          100: '#C6EDF1',
          200: '#B1E6ED',
          300: '#9CDFE8',
          400: '#88D9E3',
          500: '#73D2DE',
          600: '#4AC5D4',
          700: '#2DAEBE',
          800: '#238895',
          900: '#19626C',
        },
        'wago-lightest': '#404143',
        'wago-lightest-hover': '#666769',
        'wago-lighter': '#37383a',
        'wago-light': '#2b2c2e',
        'wago-dark': '#1A1A1A',
        'wago-darker': '#101010',
        'wago-table-row': '#262626',
        'wago-gold': '#C09100',
        'wago-muted': '#ADADAD',
        twitch: '#6441A5',
        patreon: '#F96854',
        discord: '#5865F2',
      },
      screens: {
        '3xl': '1800px',
        '4xl': '2000px',
      },
    },
  },
  variants: {
    extend: {},
  },
};

export default config;

```

**Comment by RobinMalfait** [comment]
Hey! It looks like the error you are receiving is coming from somewhere else.

We don't expose `tailwindcss/tailwind.css` in v4 anymore, can you check your codebase for instances of `@import "tailwindcss/tailwind.css";` and replace them with `@import "tailwindcss";` to see if that works?

If not, can you share a minimal reproduction repo that contains this issue so we can take a look?

Alternatively, if you are in the middle of upgrading your v3 project to v4, you could revert to the v3 state of your project, and use our experimental upgrade tool instead: `npx @tailwindcss/upgrade@next`

**Comment by Stanzilla** [comment]
> Hey! It looks like the error you are receiving is coming from somewhere else.
> 
> We don't expose `tailwindcss/tailwind.css` in v4 anymore, can you check your codebase for instances of `@import "tailwindcss/tailwind.css";` and replace them with `@import "tailwindcss";` to see if that works?
> 
> If not, can you share a minimal reproduction repo that contains this issue so we can take a look?
> 
> Alternatively, if you are in the middle of upgrading your v3 project to v4, you could revert to the v3 state of your project, and use our experimental upgrade tool instead: `npx @tailwindcss/upgrade@next`

Oh dang, my bad indeed. I need another import in my App.tsx. 

And yes, I'm indeed in the middle of a migration, I had tried the codemod before in alpha.24 but it was not working too well, it does now! 

Is there a way to use the typography and prose addons yet? The migration did nothing for them so I assume not?

**Comment by philipp-spiess** [comment]
> Is there a way to use the typography and prose addons yet? The migration did nothing for them so I assume not?

Can you add some detail as to what did not work? Both should work via the `@plugin` directive, the only thing the codemod can't do yet is to migrate any potential options you have. We can look into it if you share your Tailwind config if you want :) In either case I think this should probably be moved to a discussion since there's no actual bug in Tailwind right now (feel free to open another issue if you encounter one of course!)

**Reply by Stanzilla** [reply]
> > Is there a way to use the typography and prose addons yet? The migration did nothing for them so I assume not?
> 
> Can you add some detail as to what did not work? Both should work via the `@plugin` directive, the only thing the codemod can't do yet is to migrate any potential options you have. We can look into it if you share your Tailwind config if you want :) In either case I think this should probably be moved to a discussion since there's no actual bug in Tailwind right now (feel free to open another issue if you encounter one of course!)

Ah yep, thanks for the help. So my config is as above but I had removed the plugins line because it was complaining about it.

```js
  plugins: [require('@tailwindcss/forms'), require('@tailwindcss/typography')],
```

when putting it back and running the migration again, it does not add it to the css file, assuming because of

`│ Your configuration file could not be automatically migrated to the new CSS configuration format, so your CSS has been updated to load your existing configuration file.`

My config file is `.mts` and the require syntax no longer works there, can I import the plugins in CSS while still using the old mjs config or is there a different way?
